modules/tinymce/src/plugins/link/main/ts/core/Actions.ts

```diff
@@ -6,9 +6,16 @@ import VK from 'tinymce/core/api/util/VK';
 import * as OpenUrl from './OpenUrl';
 import * as Utils from './Utils';
 
-const getLinks = (editor: Editor) => editor.selection.isCollapsed() ?
-  Utils.getLinks(editor.dom.getParents(editor.selection.getStart())) as [HTMLAnchorElement] :
-  Utils.getLinksInSelection(editor.selection.getRng());
+const isSelectionOnImageWithEmbeddedLink = (editor: Editor) => {
+  const rng = editor.selection.getRng();
+  const node = rng.startContainer;
+  // Handle a case where an image embedded with a link is selected
+  return Utils.isLink(node) && rng.startContainer === rng.endContainer && editor.dom.select('img', node).length === 1;
```

Comment:
I'm a little surprised `getLinksInSelection` doesn't work when this image select code does 🤔 

`Utils.isLink(node) && rng.startContainer === rng.endContainer` should be enough to pass something called "get links in selection". It's a little misleading for it to only consider the selection contents, not the selection itself.

Not saying we need to change it now - but something to consider.